config/exporters/monitoring/grafana/base/prometheus-dsexporter-service.yaml

```diff
@@ -0,0 +1,112 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: appstudio-grafana-datasource-exporter
+spec: {}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: availability-exporter-sa
+  namespace: appstudio-grafana-datasource-exporter
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: availability-exporter-sa
+  namespace: appstudio-grafana-datasource-exporter
+  annotations:
+    kubernetes.io/service-account.name: availability-exporter-sa
+type: kubernetes.io/service-account-token
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: availability-exporter-cluster-role
+rules:
+- nonResourceURLs:
+  - /metrics
+  verbs:
+  - get
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: availability-exporter-cluster-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: availability-exporter-cluster-role
+subjects:
+- kind: ServiceAccount
+  name: availability-exporter-sa
+  namespace: appstudio-grafana-datasource-exporter
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: availability-exporter-service
+  namespace: appstudio-grafana-datasource-exporter
+  labels:
+    app: grafana-datasource-exporter
+spec:
+  ports:
+  - name: https
+    port: 8443
+    targetPort: https
+  selector:
+    app: grafana-datasource-exporter
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: availability-exporter-service-deployment
+  namespace: appstudio-grafana-datasource-exporter
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: grafana-datasource-exporter
+  template:
+    metadata:
+      labels:
+        app: grafana-datasource-exporter
+    spec:
+      name: availability-exporter-sa
+      containers:
+      - name: grafana-datasource-exporter
+        image: exporter:latest
+        image: quay.io/redhat-appstudio/o11y-prometheus-exporters:v0.1
+        ports:
+        - containerPort: 8443
+          name: https
+        resources:
+          limits:
+            cpu: 100m
+            memory: 100Mi
+          requests:
+            cpu: 100m
+            memory: 10Mi
+        securityContext:
+          allowPrivilegeEscalation: false
+          readOnlyRootFilesystem: true
+          runAsNonRoot: true
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: availability-metrics-reader
+  namespace: appstudio-grafana-datasource-exporter
+spec:
+  endpoints:
+  - path: /metrics
+    port: https
+    scheme: https
+    bearerTokenSecret:
+      name: "availability-exporter-sa"
```

Comment:
We are using the service account that was defined and used in the deployment.
We should create a new service account for the servicemonitor

Reply:
@avi-biton So, now the new service account will be used for the Secret and Service monitor?
Service & Deployment uses the `exporter-sa` service account